provisioning/prometheusrule.yaml

```diff
@@ -0,0 +1,114 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: metal3-ironic-prometheus-exporter-defaults
+  namespace: openshift-machine-api
+spec:
+  groups:
+  - name: baremetal.health
+    rules:
+    # Temperature alerts
+    - alert: BaremetalTemperatureWarning
```

Comment:
Something that I missed in my first review... It would be more convenient to remove the severity suffix from the alert name so rename both `BaremetalTemperatureCritical` and `BaremetalTemperatureWarning` to `BaremetalTemperature`. One advantage is that with the default Alertmanager inhibition rules, critical BaremetalTemperature alerts would automatically mute warning alerts with the same label set. It's a pattern often used upstream and in OCP too.